### PR TITLE
TE-8441: Add Specification sniffer

### PR DESCRIPTION
**Pull request by qlmmlp**
#### Overview

Ticket: [TE-8441](https://spryker.atlassian.net/browse/TE-8441)

Suite-nonsplit PR:
 - [4962](https://github.com/spryker/suite-nonsplit/pull/4962)

### CODE REVIEW

File: Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php [hunk 1]
```diff
@@ -223,35 +225,130 @@ protected function isConstructor(File $phpCsFile, int $stackPointer): bool
...
+            return;
+        }
+
+        if ($isInterface && !$specificationPosition) {
+            $phpCsFile->addErrorOnLine('Cannot fix missing specification for API interface', $tokens[$stackPointer]['line'], 'SpecificationNotFixable');
+            return;
+        }
+
+        if (!$isInterface && $specificationPosition) {
+           $phpCsFile->addErrorOnLine(static::SPECIFICATION . ' can be used only for interfaces', $tokens[$specificationPosition]['line'], 'SpecificationNotFixable');
```
Comment (gechetspr) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php:269]: Would be nice to use `sprintf` for places where you need to prepare string.

Also in this case we might want to wrap Specification: in codefences

**Review by gechetspr — CHANGES_REQUESTED** [review]
One small comment. The rest looks good for me for now

File: Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php (hunk 1 above)
Comment (dereuromark) [reply Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php:269]: Definitly ` code fencing :)

Comment (qlmmlp) [reply Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: Addressed.

**Comment by dereuromark** [comment]
After CS fixes and build is green, looks good.
Note: it might not be fully green as that is only fixed on mater. So it is fine if related changes to this PR are "green" only.

**Comment by dereuromark** [comment]
Will be merged then end of this week (TH/FR).

**Review by gechetspr — APPROVED** [review]

**Comment by dereuromark** [comment]
Interfaces do not automatically get the TODO text do they?

```
FILE: ...yker/Zed/CodeGenerator/Business/CodeGeneratorFacadeInterface.php
----------------------------------------------------------------------
FOUND 6 ERRORS AFFECTING 6 LINES
----------------------------------------------------------------------
 19 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
 28 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
 37 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
 46 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
 55 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
 64 | ERROR | Cannot fix missing specification for API
    |       | (Spryker.Commenting.DocBlockApiAnnotation.SpecificationNotFixable)
----------------------------------------------------------------------
```

That would be nice as right now it is a lot of manual work to do that. And there is no way around it now, since it fails quite hard.

**Comment by dereuromark** [comment]
It should also skip `@deprecated` ones as here it is just annoying to have to fill them as well.

**Pull request merged**